### PR TITLE
Issue 487 hybrid ids

### PR DESCRIPTION
**Pull request by JennyLouise**
### Issue reference / description

The branch you merge from should already reference an event-formation-unit github ticket number. You can add a descriptive title, but if an issue is referenced, you don't have to.

## Checklist for submitter

- [ ] Check for conflict with integration test
- [ ] Unit tests pass

---

## Nominate for Group Code Review (Anyone can nominate it)
Indicate if you think the code should be reviewed in a Thursday code review session.

- [ ] Recommend for group code review

Also, nominate it on the code_review Slack channel.

### CODE REVIEW

File: src/common/readout/vmm3/VMM3Config.cpp [hunk 1]
```diff
@@ -43,10 +132,18 @@ void VMM3Config::loadAndApplyCalibration(std::string CalibFile) {
...
     std::string HybridId = Calibration["VMMHybridCalibration"]["HybridId"];
+    if (!validHybridId(HybridId)){
+      throw std::runtime_error("Invalid HybridID in Calibration file");
+    }
     applyCalibration(HybridId, Calibration);
   }
 }
 
+bool VMM3Config::validHybridId(std::string HybridID){
+  std::regex HybridIdRegex ("E55[0-9]{29}");
```
Comment (JennyLouise) [inline src/common/readout/vmm3/VMM3Config.cpp:143]: Just check it's a 32 length string
Remove import of regex

Comment (JennyLouise) [inline src/common/readout/vmm3/VMM3Config.cpp:143]: Just check it's a 32 length string
Remove import of regex

File: src/modules/cspec/geometry/Config.h [hunk 2]
```diff
@@ -40,19 +40,12 @@ class Config : public VMM3Config {
...
-    uint32_t MaxPulseTimeNS{5 * 71'428'571}; // 5 * 1/14 * 10^9=
-    uint32_t TimeBoxNs{0xffffffff};
-    uint16_t SizeX{384};
-    uint16_t SizeY{140};
-    uint16_t SizeZ{16};
-    uint16_t MaxGridsSpan{3};
-    uint16_t DefaultMinADC{50};
-  } Parms;
-
+  std::string InstrumentGeometry{"CSPEC"};
```
Comment (JennyLouise) [inline src/modules/cspec/geometry/Config.h:43]: Consider putting the variables obtained from json file into a struct

Comment (JennyLouise) [inline src/modules/cspec/geometry/Config.h:43]: Consider putting the variables obtained from json file into a struct

File: src/common/readout/vmm3/VMM3Config.cpp (hunk 1 above)
Comment (mortenjc) [reply src/common/readout/vmm3/VMM3Config.cpp:143]: ...

**Review by mortenjc — APPROVED** [review]

**Pull request merged**